src/app/game-state/inventory.service.ts

```diff
@@ -1097,6 +1097,15 @@ export class InventoryService {
     return itemValue;
   }
 
+  /** returns false only for equipment with 0 durability */
+  hasDurability(itemStack: ItemStack): boolean {
+    const item = itemStack.item;
+
+    if (!instanceOfEquipment(item)) return true;
+
+    return (item.armorStats?.durability !== 0 && item.weaponStats?.durability !== 0);
+  }
```

Comment:
Something about this logic concerns me, but I can't pinpoint it right now. Will take another look after I get some caffeine.

Is it supposed to be `||` instead of `&&`?